activejob/lib/active_job/queue_adapters/delayed_job_adapter.rb

```diff
@@ -27,6 +27,24 @@ def enqueue_at(job, timestamp) #:nodoc:
         delayed_job
       end
 
+      def provider_job_id(job) #:nodoc:
+        begin
+          job_id_aj = job.job_id
+          djs = scan_backend(job_id_aj)
+          djs.map do |dj|
+            obj = dj.payload_object
+            next if obj.blank? || obj.job_data.blank?
+
+            job_id_persisted = obj.job_data["job_id"]
+            return dj.id if job_id_persisted == job_id_aj
+          end
+        rescue
```

Comment:
Why this rescue here?

Reply:
In the original commit, I suspect it was primarily intended to rescue the case that the backend (in that commit, ActiveRecord only) raises an exception when no records are found.

However, now that the backend access has been abstracted into its own method, the rescue should be moved to that method (even though (1) ActiveRecord appears to only return an empty collection when no records are found, and (2) ActiveRecord lazy queries are not executed within that method; perhaps future supported backends will raise such an exception)

I have committed a change that moves the rescue, although I will defer to your judgement.